addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/connector/RFXComStreamReader.java

```diff
@@ -37,89 +36,39 @@ public RFXComStreamReader(RFXComBaseConnector connector, InputStream in) {
     }
 
     @Override
-    public void interrupt() {
-        interrupted = true;
-        super.interrupt();
-        try {
-            in.close();
-        } catch (IOException ignore) {
-            // quietly close
-        }
-    }
-
-    @Override
     public void run() {
-        final int dataBufferMaxLen = Byte.MAX_VALUE;
-
-        byte[] dataBuffer = new byte[dataBufferMaxLen];
-
-        int msgLen = 0;
-        int index = 0;
-        boolean startFound = false;
-
         logger.debug("Data listener started");
 
-        try {
-
-            byte[] tmpData = new byte[20];
-            int len = -1;
-
-            while (!interrupted) {
-
-                if ((len = in.read(tmpData)) > 0) {
-
-                    byte[] logData = Arrays.copyOf(tmpData, len);
-                    logger.trace("Received data (len={}): {}", len, DatatypeConverter.printHexBinary(logData));
+        while (!Thread.interrupted()) {
+            try {
+                byte[] buf = new byte[Byte.MAX_VALUE];
 
-                    for (int i = 0; i < len; i++) {
+                if (in.read(buf, 0, 1) > 0 && buf[0] > 0) {
+                    int i = 1;
+                    int intCount = 0;
 
-                        if (index > dataBufferMaxLen) {
-                            // too many bytes received, try to find new
-                            // start
-                            startFound = false;
-                        }
-
-                        if (!startFound && tmpData[i] > 0) {
-
-                            startFound = true;
-                            index = 0;
-                            dataBuffer[index++] = tmpData[i];
-                            msgLen = tmpData[i] + 1;
-
-                        } else if (startFound) {
-
-                            dataBuffer[index++] = tmpData[i];
-
-                            if (index == msgLen) {
-                                sendMessage(dataBuffer, msgLen);
+                    while (i <= buf[0]) {
+                        int n = in.read(buf, i, buf[0] - i + 1);
 
-                                // find new start
-                                startFound = false;
-                            }
+                        if (n > 0) {
+                            i += n;
+                        } else if (intCount++ == 4) {
+                            throw new IOException("Timeout during packet read");
                         }
                     }
-                } else {
-                    try {
-                        Thread.sleep(100);
-                    } catch (InterruptedException ignore) {
-                    }
+
+                    connector.sendMsgToListeners(Arrays.copyOfRange(buf, 0, buf[0] + 1));
```

Comment:
No. It just adds nothing useful. All it did was copy the packet out of the buffer and hand it to the connector's method. We can do that ourselves without the bloat.